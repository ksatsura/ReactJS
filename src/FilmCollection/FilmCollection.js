//imports
import React from 'react';
import FilmAsset from '../FilmAsset/FilmAsset';
import '../style.css';
//constants
const key = 'qwert1';

const FilmCollection = (props) => {

    const { films, filmsNumber, filmsSameGenre, isFilmClicked } = props;

    const filmCollection = isFilmClicked ? filmsSameGenre : films;

    const collection = ( <ul className='film-collection' >
        { filmCollection.map((film, i) => <FilmAsset key={key+i} film={film} />) }
    </ul>);

    return (
        <div className='films-container'>
            { filmsNumber > 0 ? collection : <p className='no-films'>No films found</p> }
        </div>
    );   
};

export default FilmCollection;