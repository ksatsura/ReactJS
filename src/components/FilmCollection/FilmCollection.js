//imports
import * as React from 'react';
import FilmAsset from '../FilmAsset/FilmAsset';
import '../../style.css';
//constants
const key = 'qwert1';

// @flow
type Props = {
    films: Array<{}>,
    noResults: boolean,
};

const FilmCollection = (props: Props) => {

    const { films, noResults } = props;

    const filmCollection = !noResults && (<ul className='film-collection' >
        { films.map((film, i) => <FilmAsset key={key+i} film={film} /> )}
    </ul>);

    return (
        <div className='films-container'>
            { (noResults || films.length > 0)
                ? filmCollection
                : <p className='no-films'>No films found</p> 
            }
        </div>
    );   
};

const mapDispatchToProps = dispatch => {
    return {
        handleUrlChange: (genre, id) => { dispatch(fetchRequestIfNeeded(genre, true, id)); },
    };
};

export default FilmCollection;