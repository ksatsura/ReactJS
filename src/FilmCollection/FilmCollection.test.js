import React from 'react';
import FilmCollection from './FilmCollection';
import {shallow} from 'enzyme';

describe('FilmCollection.js', () => {
    const props = {
        assets: [{}, {}],
        assetsNumber: 1,
        assetsSameGenre: [{}, {}],
        isAssetClicked: false,
    };

    describe('renders', () => {

        it('renders search collection for desktop', () => {
            const component = shallow(<FilmCollection {...props} />);

            expect(component).toMatchSnapshot();
        });

        it('renders search collection for mobile', () => {
            const component = shallow(<FilmCollection {...props} isMobile />);

            expect(component).toMatchSnapshot();
        });

        it('renders genre collection for desktop', () => {
            const otherProps = {
                ...props,
                isAssetClicked: true,
            };
            const component = shallow(<FilmCollection {...otherProps} />);

            expect(component).toMatchSnapshot();
        });

        it('renders genre collection for mobile', () => {
            const otherProps = {
                ...props,
                isAssetClicked: true,
            };
            const component = shallow(<FilmCollection {...otherProps} isMobile />);

            expect(component).toMatchSnapshot();
        });

        it('renders no collection for desktop', () => {
            const otherProps = {
                ...props,
                assetsNumber: 0,
            };
            const component = shallow(<FilmCollection {...otherProps} />);

            expect(component).toMatchSnapshot();
        });

        it('renders no collection for mobile', () => {
            const otherProps = {
                ...props,
                assetsNumber: 0,
            };
            const component = shallow(<FilmCollection {...otherProps} isMobile />);

            expect(component).toMatchSnapshot();
        });
    });
});