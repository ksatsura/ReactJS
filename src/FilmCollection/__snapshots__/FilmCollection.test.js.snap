// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCollection.js renders renders genre collection for desktop 1`] = `
<div
  className="films-container"
>
  <ul
    className="film-collection"
  >
    <Connect(FilmAsset)
      film={Object {}}
      key="qwert10"
    />
    <Connect(FilmAsset)
      film={Object {}}
      key="qwert11"
    />
  </ul>
</div>
`;

exports[`FilmCollection.js renders renders no collection for desktop 1`] = `
<div
  className="films-container"
>
  <p
    className="no-films"
  >
    No films found
  </p>
</div>
`;

exports[`FilmCollection.js renders renders no collection for mobile 1`] = `
<div
  className="films-container"
>
  <p
    className="no-films"
  >
    No films found
  </p>
</div>
`;

exports[`FilmCollection.js renders renders search collection for desktop 1`] = `
<div
  className="films-container"
>
  <ul
    className="film-collection"
  >
    <Connect(FilmAsset)
      film={Object {}}
      key="qwert10"
    />
    <Connect(FilmAsset)
      film={Object {}}
      key="qwert11"
    />
  </ul>
</div>
`;
